src/components/Calculator.js

```diff
@@ -1,77 +1,102 @@
-    this.setState(
-      (state) => ({
-        dataObj: calculate(state.dataObj, buttonValue),
-      }),
-      function () {
-        let CalculatorScreen;
+  useEffect(() => {
+    let calculatorScreen;
+
+    if (dataObj.next == null && dataObj.total == null) {
+      calculatorScreen = 0;
+    } else if (dataObj.next) {
+      calculatorScreen = dataObj.next;
+    } else {
+      calculatorScreen = dataObj.total;
+    }
 
-        if (this.state.dataObj.next == null && this.state.dataObj.total == null) {
-          CalculatorScreen = 0;
-        } else if (this.state.dataObj.next) {
-          CalculatorScreen = this.state.dataObj.next;
-        } else {
-          CalculatorScreen = this.state.dataObj.total;
-        }
+    setCalculatorScreen(calculatorScreen);
+  }, [dataObj]);
 
-        this.setState({
-          CalculatorScreen,
-        });
-      },
-    );
+  const handleClick = (event) => {
+    const buttonValue = event.target.innerText;
+    setDataObj(calculate(dataObj, buttonValue));
   };
 
-  render() {
-    return (
-      <div className="calculator-container">
-        <input type="text" value={this.state.CalculatorScreen}></input>
-        <table>
-          <tr>
-            <td className="bg-light-gray" onClick={this.handleClick}>AC</td>
-            <td className="bg-light-gray" onClick={this.handleClick}>+/-</td>
-            <td className="bg-light-gray" onClick={this.handleClick}>%</td>
-            <td className="bg-orange" onClick={this.handleClick}>÷</td>
-          </tr>
-          <tr>
-            <td className="bg-light-gray" onClick={this.handleClick}>7</td>
-            <td className="bg-light-gray" onClick={this.handleClick}>8</td>
-            <td className="bg-light-gray" onClick={this.handleClick}>9</td>
-            <td className="bg-orange" onClick={this.handleClick}>x</td>
-          </tr>
-          <tr>
-            <td className="bg-light-gray" onClick={this.handleClick}>4</td>
-            <td className="bg-light-gray" onClick={this.handleClick}>5</td>
-            <td className="bg-light-gray" onClick={this.handleClick}>6</td>
-            <td className="bg-orange" onClick={this.handleClick}>-</td>
-          </tr>
-          <tr>
-            <td className="bg-light-gray" onClick={this.handleClick}>1</td>
-            <td className="bg-light-gray" onClick={this.handleClick}>2</td>
-            <td className="bg-light-gray" onClick={this.handleClick}>3</td>
-            <td className="bg-orange" onClick={this.handleClick}>+</td>
-          </tr>
-          <tr>
-            <td className="bg-light-gray" colSpan="2" onClick={this.handleClick}>0</td>
-            <td className="bg-light-gray" onClick={this.handleClick}>.</td>
-            <td className="bg-orange" onClick={this.handleClick}>=</td>
-          </tr>
-        </table>
-      </div>
-    );
-  }
+  return (
+    <div className="calculator-container">
+      <input type="text" value={calculatorScreen}></input>
```

Comment:
- [ ] React is complaining here in the console because you have defined a value property to an input field without adding an `onChange` handler. Notice that when you add a value prop to a form filed it becomes a [controlled component](https://reactjs.org/docs/forms.html) and therefore you need to add an `onChange` handler to it.

Let's fix that issue either by adding the handler or changing the `input` tag to another one

![Screenshot from 2021-09-25 11-31-01](https://user-images.githubusercontent.com/40503704/134768354-9b4f1512-3413-47f9-ba46-1486aca4b083.png)